spec.html

```diff
@@ -6614,6 +6616,41 @@ <h1>Agents</h1>
       <p>An agent is a specification mechanism and need not correspond to any particular artefact of an ECMAScript implementation.</p>
     </emu-note>
 
+    <emu-clause id="sec-agent-behaviour">
```

Comment:
I find this section confusing, and I'm not sure what it adds. I'm not sure what's missing from the descriptions in those algorithms to understand that you can call them this way.

Reply:
> I find this section confusing,

I'd like it to be less confusing. I'm not sure it can be. Do you have any suggestions?

> and I'm not sure what it adds. I'm not sure what's missing from the descriptions in those algorithms to understand that you can call them this way.

It's not so much that you *can* call them this way but that you *must* call them this way. Or at least, you must if you want the ES implementation to do anything sensible. 

The spec doesn't spend much effort to say how you *can* call an operation. For the most part, that's okay, because what matters more (at least from the *reader's* point of view) is how it *is* called, which they can see elsewhere in the spec. But, in PR #1597, these 6 operations become top-level operations, and there's zero information about how they're invoked.

If you don't single out these 6 operations, then are they on the same footing as every other operation? Is the ES spec just a box full of thousands of parts, that the host can use however it wants?